clang/test/SemaSYCL/num_simd_work_items_device.cpp

```diff
@@ -33,6 +33,68 @@ struct FuncObj {
   [[intel::num_simd_work_items(42)]] void operator()() const {}
 };
 
+#ifdef TRIGGER_ERROR
+struct TRIFuncObjBad1 {
+   [[intel::num_simd_work_items(3)]] // expected-error{{'num_simd_work_items' attribute conflicts with ''reqd_work_group_size'' attribute}}
```

Comment:
Should we consider adding a new diagnostic (or modifying the existing one) for this case to explain why they conflict?

Reply:
I am in favor of adding new diagnostic so that we can clearly explain why they conflict here. I will add it. Thanks for pointing this.

Reply:
I have added new diagnostic here.